### DISCUSSION TITLE
Time explicit in the objective function

### DISCUSSION THREAD

**Discussion by Peter230655** [csu-hmc/opty#258] [Q&A]
If the time t explicitly appears in the equations of motion, the example plot_betts_2003 in examples-gallery shows how to handle this:
A sympy function, say, T(t) is used and it replaces the time variable t whereever it is explicit in the eoms. In the problem it becomes a known_trajectories_map.
Now I have the time t explicitly on the objective function as: $\int [(y_N - 2 + e^{\rho \cdot t})^2 +\gamma \cdot v_1^2 ] \ dt$, where $\rho, \gamma$ are constants which appear in the objective function with their numerical value, $y_N$ is a state variable and $v_1$ is a control variable.

Can opty.utils.create_objective_function handle this explicit dependency on t, if yes how to do it?

**Thanks for any help!**

**Comment by Peter230655** [comment]
If I understood @tjstienstra correctly, currently opty.utils.create_objective_function cannot handle explicit time dependence.

**Reply by tjstienstra** [reply]
The equations and the code should at least be checked to verify whether it would support it or can be modified to support it.

**Reply by Peter230655** [reply]
> The equations and the code should at least be checked to verify whether it would support it or can be modified to support it.

I agree, it is such a convenient function!! But I understood correctly, that right now it will not support explicit time dependence?

**Reply by tjstienstra** [reply]
As you can see [here](https://github.com/csu-hmc/opty/blob/aae009e48f1c1e9d19f43b12c650929622b0a510/opty/utils.py#L332) it doesn't take into account that it might have to substitute `t` so it doesn't support it.

**Reply by tjstienstra** [reply]
However, if this issue were to be tackled, it should be done i.c.w. #190. Seems that I have written a proposal for it already some time ago. It mainly requires some verification of the math.
